extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -46,37 +46,54 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 				.withAdditionalProperties(propertyBag).send();
 
 			// prompt user for connection profile to get connection info
-			quickPickStep = 'getConnectionInfo';
-			connectionInfo = await vscodeMssqlApi.promptForConnection(true);
-			if (!connectionInfo) {
-				// User cancelled
-				return;
-			}
-			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
-				.withAdditionalProperties(propertyBag).withConnectionInfo(connectionInfo).send();
-
-			// list databases based on connection profile selected
-			connectionURI = await vscodeMssqlApi.connect(connectionInfo);
-			listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
-			const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
-				canPickMany: false,
-				title: constants.selectDatabase,
-				ignoreFocusOut: true
-			}));
+			while (true) {
+				connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+				if (!connectionInfo) {
+					// User cancelled
+					return;
+				}
+				quickPickStep = 'getConnectionInfo';
+				try {
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
+							title: constants.connectionProgressTitle,
+							cancellable: false
+						}, async (_progress, _token) => {
+							// list databases based on connection profile selected
+							if (!connectionInfo) {
```

Comment:
Aren't you already checking for connectionInfo above?

Reply:
If I don't include this it shows me this error within the withProgress: 
<img width="437" alt="image" src="https://user-images.githubusercontent.com/23587151/165371349-0eb8b67e-6c78-46fc-ab36-ba80d0aae4b8.png">

Reply:
Probably because it's in a lambda function. We know it's going to exist here though so you can just do `vscodeMssqlApi.connect(connectionInfo!)`